R/addWaterYear.R

```diff
@@ -36,35 +38,45 @@ addWaterYear <- function(rawData){
   # set the name of the date column(s) to use for calculating the WY
   dateColName <- names(rawData)[names(rawData) %in% allowedDateColNames]
   
-  # set up new df that will include water year columns
-  wyData <- rawData
-  
   for(dateCol in dateColName){
-    dateColWY <- paste0(dateCol, "WY")
+    dateColWY <- allowedWYColNames[dateCol]
     
     # if this WY column already exists, do not add another
-    if(dateColWY %in% names(wyData)){
+    if(dateColWY %in% names(rawData)){
```

Comment:
Wait, a little confused here. I am already doing the above lines you provided with `dateColName <- names(rawData)[names(rawData) %in% allowedDateColNames]` which only returns the column name to loop over that exists in `rawData`. I need to replace the if/next which deals with the WY col names. So following the same pattern-ish but for WY col names.